packages/medusa-payment-stripe/src/helpers/stripe-base.js

```diff
@@ -99,49 +95,117 @@ class StripeBase extends AbstractPaymentService {
    * @return {Promise<object>} Stripe customer
    */
   async retrieveCustomer(customerId) {
-    return await this.stripeProviderService_.retrieveCustomer(customerId)
-  }
-
-  /**
-   * Creates a Stripe customer using a Medusa customer.
-   * @param {object} customer - Customer data from Medusa
-   * @return {Promise<object>} Stripe customer
-   */
-  async createCustomer(customer) {
-    return await this.stripeProviderService_
-      .withTransaction(this.manager_)
-      .createCustomer(customer)
+    if (!customerId) {
+      return
+    }
+    return await this.stripe_.customers.retrieve(customerId)
   }
 
   /**
    * Creates a Stripe payment intent.
    * If customer is not registered in Stripe, we do so.
-   * @param {Cart} cart - cart to create a payment for
-   * @return {Promise<PaymentSessionData>} Stripe payment intent
+   * @param {Cart & PaymentContext} context - context to use to create a payment for
+   * @return {Promise<PaymentSessionResponse>} Stripe payment intent
    */
-  async createPayment(cart) {
+  async createPayment(context) {
     const intentRequestData = this.getPaymentIntentOptions()
+    const { id: cart_id, email, context: cart_context, collected_data, currency_code, amount, resource_id } = context
+
+    if (!isDefined(currency_code) || !isDefined(amount)) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        'Unable to create a new payment, one of the "currency_code" or "amount" values is not provided. Both values are required.'
+      )
+    }
+
+    const intentRequest = {
+      description:
+        cart_context.payment_description ??
+        this.options_?.payment_description,
+      amount: Math.round(amount),
+      currency: currency_code,
+      metadata: { cart_id, resource_id },
+      capture_method: this.options_.capture ? "automatic" : "manual",
+      ...intentRequestData,
+    }
 
-    return await this.stripeProviderService_
-      .withTransaction(this.manager_)
-      .createPayment(cart, intentRequestData)
+    if (this.options_?.automatic_payment_methods) {
+      intentRequest.automatic_payment_methods = { enabled: true }
+    }
+
+    if (collected_data?.stripe_id) {
```

Comment:
thought: Think we should send `stripe_customer_id` here to avoid ambiguity

Reply:
I could, but it would break any other existing payment plugins relying on stripe. Wdyt? A migration which duplicate the existing key with the new naming could help?

Reply:
Could we do `metadata?.stripe_id ?? metadata?.stripe_customer_id` ?

Reply:
We are not in control of the customisations or plugins made outside of medusa. It would be a breaking changes. I tried in that pr to have no breaking changes while introducing the new api. From the steps i have shared in the rfc, i proposed the breaking changes in a separate pr, when we will be ready to merge the interface and abstract class i ve proposed wdyt?

Reply:
Makes sense. Let's keep it as is :)